src/otaclient/ota_core/_updater.py

```diff
@@ -178,12 +177,14 @@ def _execute_update(self):
 
         self._handle_upper_proxy()
         self._process_metadata()
-        self._pre_update()
+        with self.critical_zone_flags:
+            self._pre_update()
         _delta_digests = self._calculate_delta()
         self._download_delta_resources(_delta_digests)
         self._apply_update()
-        self._post_update()
-        self._finalize_update()
+        with self.critical_zone_flags:
+            self._post_update()
+            self._finalize_update()
```

Comment:
will be better like the following to interrupt stop control.
```suggestion
        with self.critical_zone_flags:
            self._post_update()
        with self.critical_zone_flags:
            self._finalize_update()
```